components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -384,11 +384,86 @@ public void deleteGatewayArtifacts(String apiId) throws APIManagementException {
         return apiRuntimeArtifactDtoList;
     }
 
+    public List<APIRuntimeArtifactDto> retrieveGatewayArtifactsByAPIIDs(List<String> apiIds, String[] labels,
+                                                                        String tenantDomain)
+            throws APIManagementException {
+        // Split apiId list into smaller list of size 25
+        List<List<String>> apiIdsChunk = new ArrayList<>();
+        int apiIdListSize = apiIds.size();
+        int apiIdArrayIndex = 0;
+        int apiIdsChunkSize = SQLConstants.API_ID_CHUNK_SIZE;
+        while (apiIdsChunkSize < apiIdListSize) {
+            apiIdsChunk.add(apiIds.subList(apiIdArrayIndex, apiIdsChunkSize));
+            apiIdListSize = apiIdListSize - SQLConstants.API_ID_CHUNK_SIZE;
+            apiIdArrayIndex = apiIdArrayIndex + SQLConstants.API_ID_CHUNK_SIZE;
+            apiIdsChunkSize = apiIdsChunkSize + SQLConstants.API_ID_CHUNK_SIZE;
+        }
+        if (apiIdListSize > 0) {
+            apiIdsChunk.add(apiIds.subList(apiIdArrayIndex, apiIdArrayIndex + apiIdListSize));
+        }
```

Comment:
Please check if you can simplify the logic here as the following example.

```java
        List<String> apiIds = Arrays.asList("id1","id2","id3","id4","id5","id6","id7");
        int PARTITION_SIZE = 3;
        int INDEX = 0;
        int ID_COUNT = apiIds.size();
        List<List<String>> apiPartitions = new ArrayList<>();
        while (INDEX < ID_COUNT) {
            apiPartitions.add(apiIds.subList(INDEX, Math.min(INDEX + PARTITION_SIZE, ID_COUNT)));
            INDEX += PARTITION_SIZE;
        }
        System.out.println(apiPartitions);
```

Output:
```
[[id1, id2, id3], [id4, id5, id6], [id7]]
```

Reply:
Fixed with b47708f01abb761d5651b988a460491bb59d2e9d